frontend/src/pages/modelRegistry/screens/utils.ts

```diff
@@ -100,21 +101,19 @@ export const getCustomPropString = <
 };
 
 export const getCatalogModelDetailsProps = (
-  customProps: ModelRegistryCustomProperties,
-): CatalogModelDetailsParams => ({
-  sourceName: getCustomPropString(customProps, '_registeredFromCatalogSourceName'),
-  repositoryName: getCustomPropString(customProps, '_registeredFromCatalogRepositoryName'),
-  modelName: getCustomPropString(customProps, '_registeredFromCatalogModelName'),
-  tag: getCustomPropString(customProps, '_registeredFromCatalogTag'),
-});
-
-export const getPipelineModelCustomProps = (
-  customProps: ModelRegistryCustomProperties,
-): PipelineModelCustomProps => ({
-  project: getCustomPropString(customProps, '_registeredFromPipelineProject'),
-  runId: getCustomPropString(customProps, '_registeredFromPipelineRunId'),
-  runName: getCustomPropString(customProps, '_registeredFromPipelineRunName'),
-});
+  artifact: ModelArtifact,
+): CatalogModelDetailsParams | null => {
+  // Check for modelSource* properties using the new standardized approach
+  if (artifact.modelSourceKind === ModelSourceKind.CATALOG && artifact.modelSourceName) {
+    return {
+      sourceName: artifact.modelSourceClass || '',
+      repositoryName: artifact.modelSourceGroup || '',
+      modelName: artifact.modelSourceName,
+      tag: artifact.modelSourceId || '',
+    };
+  }
+  return null;
+};
```

Comment:
I think this function is redundant with your `modelSourcePropertiesToCatalogParams` util, you can still just pass that function an artifact and it'll work because an artifact contains the `ModelSourceProperties` (In typescript terms, ModelArtifact is assignable to ModelSourceProperties). You can get rid of `getCatalogModelDetailsProps` and in the places where it was used, just drop in `modelSourcePropertiesToCatalogParams`.